storage_delegation/main.tf

```diff
@@ -6,16 +12,48 @@ locals {
     "eu-de"    = "eu-de.dataplatform.cloud.ibm.com",
     "jp-tok"   = "jp-tok.dataplatform.cloud.ibm.com"
   }
-  dataplatform_ui = local.dataplatform_ui_mapping[local.location]
+  dataplatform_ui             = local.dataplatform_ui_mapping[local.location]
+  kms_service                 = module.crn_parser.service_name
+  kms_account_id              = module.crn_parser.account_id
+  kms_key_id                  = module.crn_parser.resource
```

Comment:
all these need to use ternary logic, because the `crn_parser` module should only be called if user is passing a value for `cos_kms_key_crn`

Reply:
In case user is not passing cos_kms_key_crn parser is called on the key created by the module.